server/src/unittest/gpos/task/CAutoTaskProxyTest.cpp

```diff
@@ -539,6 +567,57 @@ CAutoTaskProxyTest::EresUnittest_PropagateExecError()
 	return GPOS_OK;
 }
 
+//---------------------------------------------------------------------------
+//	@function:
+//		CAutoTaskProxyTest::EresUnittest_ExecuteError
+//
+//	@doc:
+//		Check if we can see the exception if task and autotask run in same thread
+//		through execute
+//
+//---------------------------------------------------------------------------
+GPOS_RESULT
+CAutoTaskProxyTest::EresUnittest_ExecuteError()
+{
+	CAutoMemoryPool amp;
+	IMemoryPool *pmp = amp.Pmp();
+
+	// Create new thread so worker for this new
+	// thread is available to run the task ("CAutoTaskProxyTest::PvUnittest_Error")
+	// from Unittest_CheckExecuteErrorInternal
+	STestThreadDescriptor st;
+	st.ulId = GPOS_THREAD_MAX + 1;
+	st.m_pmp = pmp;
+	st.fException = false;
+	st.fPropagateException = true;
+
+	INT res = pthread_create(&st.m_pthrdt, NULL /*pthrAttr*/, Unittest_CheckExecuteErrorInternal, &st);
+
+	// check for error
+	if (0 != res)
+	{
+		return GPOS_FAILED;
+	}
+	pthread_join(st.m_pthrdt, NULL);
+	if (!st.fException)
+	{
+		return GPOS_FAILED;
+	}
+
+	// Disable Propagate
+	st.fPropagateException = false;
+	res = pthread_create(&st.m_pthrdt, NULL /*pthrAttr*/, Unittest_CheckExecuteErrorInternal, &st);
+
+	// check for error
+	if (0 != res)
+	{
+		return GPOS_FAILED;
+	}
+	pthread_join(st.m_pthrdt, NULL);
+	return !st.fException ? GPOS_OK : GPOS_FAILED;
```

Comment:
Well, I tried to split the statements to make it clear. Something like:

```
// Expect no propagation of exception
if (st.fException)
{
  return GPOS_FAILED;
}

return GPOS_OK;
```

Other than that, very well done. Thanks.